presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/CombineApproxPercentileFunctions.java

```diff
@@ -0,0 +1,277 @@
+                        new ArrayType(percentileArrayBuilder.build().get(0).getType()),
+                        percentileArrayBuilder.build());
+                VariableReferenceExpression arrayVariableReference = context.getVariableAllocator().newVariable(array);
+                // Add array to project node
+                childProjectAssignments.put(arrayVariableReference, array);
+
+                // Create new aggregation
+                AggregationNode.Aggregation curAgg = entry.getKey();
+                int percentilePosition = getPercentilePosition(curAgg.getCall().getFunctionHandle());
+                if (percentilePosition == -1) {
+                    return Result.empty();
+                }
+                List<RowExpression> newAggCallArguments = addPercentileArgument(curAgg.getCall().getArguments(), arrayVariableReference, percentilePosition);
+                Type colType = entry.getKey().getCall().getArguments().get(0).getType();
+                CallExpression approxPercentileCall = call(
+                        functionAndTypeManager,
+                        "approx_percentile",
+                        new ArrayType(colType),
+                        newAggCallArguments);
+
+                AggregationNode.Aggregation newAggregation =
+                        new AggregationNode.Aggregation(
+                                approxPercentileCall,
+                                curAgg.getFilter(),
+                                curAgg.getOrderBy(),
+                                curAgg.isDistinct(),
+                                curAgg.getMask());
+                // Get variable reference to the new aggregation
+                VariableReferenceExpression newVariableReference = context.getVariableAllocator().newVariable(newAggregation.getCall());
+                aggregations.put(newVariableReference, newAggregation);
+
+                int curIdx = 1;
+                for (Pair<VariableReferenceExpression, RowExpression> pair : entry.getValue()) {
+                    FunctionHandle idxHandle = functionAndTypeManager.lookupFunction("element_at", fromTypes(newVariableReference.getType(), BIGINT));
+                    RowExpression idx = new CallExpression(
+                            "element_at",
+                            idxHandle,
+                            colType,
+                            ImmutableList.of(newVariableReference, constant((long) curIdx, BIGINT)));
+                    parentProjectAssignments.put(pair.getFirst(), idx);
+                    combinedAggVariableReference.add(pair.getFirst());
+                    curIdx += 1;
+                }
+            }
+        }
+
+        if (!hasChange) {
+            return Result.empty();
+        }
+
+        // Add aggregations which are not combined to the new aggregation node.
+        for (Map.Entry<VariableReferenceExpression, AggregationNode.Aggregation> entry : aggregation.getAggregations().entrySet()) {
```

Comment:
Using forEach to iterate map is much simpler as you avoid the entry api

Reply:
Done